Add some clarification to Fronius integration docs (#18888)
home-assistant/home-assistant.github.io: source/_integrations/fronius.markdown

  - sensor
---

> The `fronius` sensor polls a [Fronius](https://www.fronius.com/) solar inverter, battery system or smart meter and presents the values as sensors in Home Assistant. Data is gathered directly through the system's datamanager card connected to your local network, this integration doesn't access the cloud. 

Reviewer: ```suggestion
The `fronius` sensor polls a [Fronius](https://www.fronius.com/) solar inverter, battery system or smart meter and presents the values as sensors in Home Assistant. Data is gathered directly from a Fronius device connected to your local network, this integration doesn't access the cloud. 
```

## Prerequisites

> You will need to either set a static IP on the datamanager card or assign a static DHCP lease for it, or alternatively access it through the local DNS name if your network is properly configured for this.

Reviewer: ```suggestion
You will need to either set a static IP on the Fronius device or assign a static DHCP lease for it, or alternatively access it through the local DNS name if your network is properly configured for this.
```

  - sensor
---

> The `fronius` sensor polls a [Fronius](https://www.fronius.com/) solar inverter, battery system or smart meter and presents the values as sensors in Home Assistant. Data is gathered directly through the system's datamanager card connected to your local network, this integration doesn't access the cloud. You will need to either set a static IP or assign a static DHCP lease, or alternatively access it through the local DNS name if your network is properly configured.

Reviewer 2: This is now an introduction and an instruction in one. Please split the prerequisites from the introduction.

Author: Done, thanks for the observation.

```yaml
sensor:
  - platform: fronius
    resource: FRONIUS_LOCAL_API_URL
    monitored_conditions:
    - sensor_type: inverter
```

{% configuration %}
resource:

>   description: "The URL of the API on the Fronius datamanager card (e.g., `http://192.0.2.0` or `http://fronius.local`)"

Reviewer 2: Is the API different from regular device access? As in, as it specifically mentions it, is that needed?

Author: Clarified that only root URL is needed.

```yaml
sensor:
  - platform: fronius

>     resource: FRONIUS_CARD_LOCAL_URL

Reviewer: ```suggestion
    resource: FRONIUS_URL_OR_IP
```

    monitored_conditions:
    - sensor_type: inverter
```

{% configuration %}
resource:

>   description: "The root URL of the Fronius datamanager card (e.g., `http://192.0.2.0` or `http://fronius.local`)"

Reviewer: ```suggestion
  description: "The hostname or IP address of the Fronius device (e.g., `192.0.2.0` or `http://fronius.local`)"
```

This does not need to be problematic as the values will be added on sunrise,
when the Fronius devices begins providing the needed data.

## Finding out devices IDs

To find out the device ID of the inverter visit the URL:

> http://your_datamanager_card_ip/solar_api/v1/GetPowerFlowRealtimeData.fcgi

Reviewer 2: Don't use bare URLs, use all caps for replacement values.

Author: Done.

> In the returned json, under the key Body > Data > Inverters you should see your inverters listed with IDs starting from 1.

Reviewer 2: ```suggestion
In the returned JSON, under the key Body > Data > Inverters you should see your inverters listed with IDs starting from 1.
```

To find out the device ID of the meter visit the URL:

> http://your_datamanager_card_ip/solar_api/v1/GetMeterRealtimeData.cgi?Scope=System

Reviewer 2: Don't use bare URLs, use all caps for replacement values.

Author: Done.

> In the returned json, under the key Body > Data you should see your meters listed with IDs starting from 0.

Reviewer 2: ```suggestion
In the returned JSON, under the key Body > Data you should see your meters listed with IDs starting from 0.
```

To find out which API version your system runs, visit the URL:

> http://your_datamanager_card_ip/solar_api/GetAPIVersion.cgi

Reviewer 2: Don't use bare URLs, use all caps for replacement values

Author: Done.

This does not need to be problematic as the values will be added on sunrise,
when the Fronius devices begins providing the needed data.

## Finding out devices IDs

To find out the device ID of the inverter visit the URL:

> http://FRONIUS_IP/solar_api/v1/GetPowerFlowRealtimeData.fcgi

Reviewer 2: ```suggestion
`http://FRONIUS_IP/solar_api/v1/GetPowerFlowRealtimeData.fcgi`
```

> http://FRONIUS_DATAMANAGER_CARD_IP/solar_api/v1/GetPowerFlowRealtimeData.fcgi

Reviewer: ```suggestion
http://FRONIUS_IP/solar_api/v1/GetPowerFlowRealtimeData.fcgi
```

In the returned JSON, under the key Body > Data > Inverters you should see your inverters listed with IDs starting from 1.

To find out the device ID of the meter visit the URL:

> http://FRONIUS_IP/solar_api/v1/GetMeterRealtimeData.cgi?Scope=System

Reviewer 2: ```suggestion
`http://FRONIUS_IP/solar_api/v1/GetMeterRealtimeData.cgi?Scope=System`
```

> http://FRONIUS_DATAMANAGER_CARD_IP/solar_api/v1/GetMeterRealtimeData.cgi?Scope=System

Reviewer: ```suggestion
http://FRONIUS_IP/solar_api/v1/GetMeterRealtimeData.cgi?Scope=System
```

In the returned JSON, under the key Body > Data you should see your meters listed with IDs starting from 0.

To find out which API version your system runs, visit the URL:

> http://FRONIUS_IP/solar_api/GetAPIVersion.cgi

Reviewer 2: ```suggestion
`http://FRONIUS_IP/solar_api/GetAPIVersion.cgi`
```

> http://FRONIUS_DATAMANAGER_CARD_IP/solar_api/GetAPIVersion.cgi

Reviewer: ```suggestion
http://FRONIUS_IP/solar_api/GetAPIVersion.cgi
```

      device: 0
    - sensor_type: power_flow
```

## Note

> Fronius often provides firmware updates for the datamanager interfaces and the devices in their system, it's recommended to check and apply them regurarly.

Reviewer 2: How is this related to the integration?

Author: It is in the aspect that some parts of the API used by this integration are missing in older firmware revisions of the datamanager card. I thought it could be useful to point people in the direction of updating it to the lastest before starting to look for issues in HA.
Clarified the statement accordingly.

      device: 0
    - sensor_type: power_flow
```

## Note

> Fronius often provides firmware updates for the datamanager interfaces and the devices in their system, it's recommended to check and apply them regurarly. This integration relies on functionality present in rather recent firmware.

Reviewer 2: ```suggestion
Fronius often provides firmware updates for the datamanager interfaces and the devices in their system, it's recommended to check and apply them regularly. This integration relies on functionality present in rather recent firmware.
```
